fix(libs): fix empty type and kind column on types table (#2117)

jethrocabaluna opened the pull request:
<!-- Please make sure that your commit message follows our format -->
<!-- Example: `fix(libs-ui-component): must begin with lowercase` -->

## Description (Optional)

<!-- This is a short description on the Pull Request -->
- load sub types of current page only when expanded
- make typesList alphabetical

## Related Issue(s)

<!-- Please link the issue being fixed so it gets closed when this is merged. -->

Fixes #2117 
Fixes #2090

webberwang commented on libs/frontend/domain/type/src/store/type.service.ts:
```diff
@@ -98,7 +97,12 @@ export class TypeService
   @computed
   get typesList() {
-    return [...this.types.values()]
+    // loading sub types messes up the order of the next page
```
how come the backend sort by name isn't enough?

jethrocabaluna replied:
The sort by name on the cypher query works well until we get the types of the interfaceType fields. From what I understand from the `class TypeService`, it has an objectMap `types` property and we populate it whenever we get types. So in our pagination:
1. The types in the first page are saved on the `this.types`
2. If we get the types of the fields of an interfaceType, those types will also be saved in the `this.types`
3. When navigated on the second page, the field types that were saved appears before the new set of base types
4. We get the `this.types` with `[...this.types.values()]`, somehow this doesn't return them alphabetically  so I needed to add an extra sort
![scrnli_1_5_2023_5-10-38 PM](https://user-images.githubusercontent.com/27695022/210744566-f049d372-ba9c-423f-a5e2-5c60e26de183.png)
![scrnli_1_5_2023_5-11-51 PM](https://user-images.githubusercontent.com/27695022/210744613-abe7445e-cecb-436e-9d53-5666e0bc24b3.png)

webberwang commented on libs/frontend/domain/type/src/store/type.service.ts:
```diff
@@ -190,23 +194,9 @@ export class TypeService
-
-    const allIds = [...ids, ...(where?.id_IN || [])]
-    const types = yield* _await(getAllTypes(allIds))
+  getAll = _async(function* (this: TypeService, where?: BaseTypeWhere) {
```
I see offset is removed, but what if want offset and limit to be passed here?

jethrocabaluna replied:
The offset and limit were only used for the `this.getBaseTypes`. I removed that block of code inside the `getAll` so theres no usage of the offset and limit anymore in the function. The `getAllTypes` only accepts a single argument which is an array of string which should be the ids.

The reason I removed the usage of `this.getBaseTypes` is because it can be unnecessary when `getAll` is used but doesn't really need the baseTypes. For example is the `getOne`, which calls `getAll` with a specific id if the type doesn't exist yet in the `this.types`. The `this.getBaseTypes` doesn't need to run in the `getOne`

webberwang replied:
okay this makes sense, I think prev dev added unnecessary logic

webberwang commented on libs/frontend/domain/type/src/use-cases/types/get-types/useTypesTableData.ts:
```diff
@@ -2,13 +2,25 @@ import type { ITypeService } from '@codelab/frontend/abstract/core'
-    typeService.getAll.bind(typeService),
-    [],
+  typeService.getBaseTypes.bind(typeService)
+  typeService.getAll.bind(typeService)
```
shouldn't get all already get base types?

jethrocabaluna replied:
In this PR, I removed the use of `getBaseTypes` inside the `getAll` because I think it could run unnecessarily and we may use `getAll` but dont need the baseTypes like in the method [getOne](https://github.com/codelab-app/builder/blob/4916af3705dc046340c5095cdbe6593f4a589576/libs/frontend/domain/type/src/store/type.service.ts#L218).

Since they are separated now, I had to use `getBaseTypes` first to get the ids in the current page and use `getAll` with those ids to populate the necessary fields for the baseTypes (e.g. `fields` property of an interfaceType)

jethrocabaluna commented:
In this video, we can see that the field types are only loaded upon expanding.
@webberwang let me know if should I load the sub field types of interfaceTypes in the background instead of in-demand.
[interface-types-pagination.webm](https://user-images.githubusercontent.com/27695022/210751222-d9d5622b-c3ac-42c2-96af-2ba47ac5db1d.mp4)

jethrocabaluna commented:
[interface-types-pagination.webm](https://user-images.githubusercontent.com/27695022/210752106-490b16e9-c9e5-4b0d-ab71-e8a63386ec09.mp4)

jethrocabaluna commented:
Updated the PR so it doesn't only get the descendants of `interfaceTypes`. I moved that logic upon expanding the type.

[type-descendants.webm](https://user-images.githubusercontent.com/27695022/211058039-1ff106a7-ce55-447e-bb59-67ed89905768.webm)
